pkg/controller/migcluster/validation.go

```diff
@@ -335,3 +345,44 @@ func (r *ReconcileMigCluster) validateSaTokenPrivileges(cluster *migapi.MigClust
 	}
 	return nil
 }
+
+// validate operator version.
+func (r ReconcileMigCluster) validateOperatorVersionMatchesHost(cluster *migapi.MigCluster) error {
+	if cluster.Spec.IsHostCluster {
+		return nil
+	}
+	if cluster.Status.HasCriticalCondition() {
+		return nil
+	}
+
+	client, err := cluster.GetClient(r)
+	if err != nil {
+		return liberr.Wrap(err)
+	}
+
+	clusterConfig, err := cluster.GetClusterConfigMap(client)
```

Comment:
I think you could just use your own func `GetOperatorVersion()` here, maybe that's the plan.

Reply:
Good call